tests/end-to-end/director/application_api_test.go

```diff
@@ -0,0 +1,856 @@
+	}
+	actualAppPage := graphql.ApplicationPage{}
+
+	// WHEN
+	queryReq := gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: applications {
+					%s
+				}
+			}`, tc.gqlFieldsProvider.Page(tc.gqlFieldsProvider.ForApplication())))
+	err := tc.RunQuery(ctx, queryReq, &actualAppPage)
+	saveQueryInExamples(t, queryReq.Query(), "query applications")
+
+	//THEN
+	require.NoError(t, err)
+	assert.Len(t, actualAppPage.Data, 3)
+	assert.Equal(t, 3, actualAppPage.TotalCount)
+
+}
+
+func TestQuerySpecificApplication(t *testing.T) {
+	// GIVEN
+	in := graphql.ApplicationInput{
+		Name: fmt.Sprintf("app"),
+	}
+
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+
+	actualApp := graphql.Application{}
+	request := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+			result: createApplication(in: %s) {
+					%s
+				}
+			}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	err = tc.RunQuery(context.Background(), request, &actualApp)
+	require.NoError(t, err)
+	require.NotEmpty(t, actualApp.ID)
+	createdID := actualApp.ID
+	defer deleteApplication(t, actualApp.ID)
+
+	// WHEN
+	queryAppReq := gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: application(id: "%s") {
+					%s
+				}
+			}`, actualApp.ID, tc.gqlFieldsProvider.ForApplication()))
+	err = tc.RunQuery(context.Background(), queryAppReq, &actualApp)
+	saveQueryInExamples(t, queryAppReq.Query(), "query specific application")
+
+	//THEN
+	require.NoError(t, err)
+	assert.Equal(t, createdID, actualApp.ID)
+}
+
+func TestTenantSeparation(t *testing.T) {
+	//TODO later
+}
+
+func getApp(ctx context.Context, t *testing.T, id string) ApplicationExt {
+	q := gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: application(id: "%s") {
+				%s
+			} 
+		}`, id, tc.gqlFieldsProvider.ForApplication()))
+	var app ApplicationExt
+	require.NoError(t, tc.RunQuery(ctx, q, &app))
+	return app
+
+}
+
+func generateSampleApplicationInput(placeholder string) graphql.ApplicationInput {
+	return graphql.ApplicationInput{
+		Name: placeholder,
+		Documents: []*graphql.DocumentInput{{
+			Title:  placeholder,
+			Format: graphql.DocumentFormatMarkdown}},
+		Apis: []*graphql.APIDefinitionInput{{
+			Name:      placeholder,
+			TargetURL: placeholder}},
+		EventAPIs: []*graphql.EventAPIDefinitionInput{{
+			Name: placeholder,
+			Spec: &graphql.EventAPISpecInput{
+				EventSpecType: graphql.EventAPISpecTypeAsyncAPI,
+				Format:        graphql.SpecFormatYaml,
+			}}},
+		Webhooks: []*graphql.ApplicationWebhookInput{{
+			Type: graphql.ApplicationWebhookTypeConfigurationChanged,
+			URL:  placeholder},
+		},
+		Labels:      &graphql.Labels{placeholder: []string{placeholder}},
+		Annotations: &graphql.Annotations{placeholder: placeholder},
+	}
+}
+
+func deleteApplication(t *testing.T, id string) {
+	req := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+		deleteApplication(id: "%s") {
+			id
+		}	
+	}`, id))
+	require.NoError(t, tc.RunQuery(context.Background(), req, nil))
```

Comment:
I would still split this file into separate files under `application` domain. But anyway, we can do it later.